test/integration/groups_test.go

```diff
@@ -68,11 +71,28 @@ var _ = Describe("Groups", func() {
 			}
 			Expect(found).To(BeTrue())
 
-			// get group by name
+			// create new cluster
+			newClusterDetails, err := c.Clusters.RegisterCluster(testConfig.OrgID, types.Registration{Name: clusterName})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(newClusterDetails).NotTo(BeNil())
+
+			// add new cluster to the group
+			_, err = c.Groups.GroupClusters(testConfig.OrgID, newGroupDetails.UUID, []string{newClusterDetails.ClusterID})
+			Expect(err).NotTo(HaveOccurred())
+
+			// get group by name and verify that the new cluster is part of it
```

Comment:
Could you possibly try to put all this new stuff into its own test? I see why you didn't, given that there's currently just this one megatest. I'm honestly not a huge fan of that, though, it would be interesting to see if it's possible to break it up into smaller tests. Your new functionality could be the guinea pig for that :smile: thanks!

Reply:
You're right. That would make it easier to check which part actually failed.